st2common/st2common/services/executions.py

```diff
@@ -132,9 +142,12 @@ def _get_parent_execution(child_liveaction_db):
 def update_execution(liveaction_db, publish=True):
     execution = ActionExecution.get(liveaction__id=str(liveaction_db.id))
     decomposed = _decompose_liveaction(liveaction_db)
+    kw = {}
     for k, v in six.iteritems(decomposed):
-        setattr(execution, k, v)
-    execution = ActionExecution.add_or_update(execution, publish=publish)
+        kw['set__' + k] = v
```

Comment:
`persistence` is for all DB specific abstractions. Limits itself to a single model type like `ActionDB` etc. In theory replacing MongoDB with say Postgres should have required the bulk of the changes to be under the `persistence` layer only however due to our failure to maintain abstractions faithfully that will not be true. 

`service` - provides a higher level feature which can use persistence to retrieve data from the DB. Works simultaneously with multiple models and generally capture the meat of the what has been traditionally thought of as "business logic". https://github.com/StackStorm/st2/blob/master/st2common/st2common/services/action.py#L47 is a pretty decent example of what comprises service and why it is not necessary part of persistence.

Often abstraction do not make sense in all contexts however the lack of consistent abstractions make the code base challenging to reason about which is why I personally prefer to maintain them even if they might seem superfluous.